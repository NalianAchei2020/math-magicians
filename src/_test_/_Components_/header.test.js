import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import { render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Header from '../../components/header/header';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('testing for the UI of Navbar', () => {
  act(() => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
      container
    );
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
