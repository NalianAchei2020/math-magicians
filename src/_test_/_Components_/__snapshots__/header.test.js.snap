// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for the UI of Navbar 1`] = `""`;
